ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Protocol.hs

```diff
@@ -412,7 +412,11 @@ instance TPraosCrypto era => ConsensusProtocol (TPraos era) where
               -> Nothing
     where
       chainState = tickedTPraosStateChainDepState cs
-      lv         = getTickedPraosLedgerView (tickedTPraosStateLedgerView cs)
+      lv         = getTickedPraosLedgerView $ tickedTPraosStateLedgerView cs
+      d          = SL._d $ SL.lvProtParams lv
+      asc        = tpraosLeaderF $ tpraosParams cfg
+      firstSlot  = firstSlotOfEpochOfSlot (tpraosEpochInfo cfg) slot
+      gkeys      = Map.keys dlgMap
```

Comment:
Does the order of the `gkeys` list matter? Is _ascending_ correct for this use in `cardano-ledger-specs:...lookupIn...`?

Reply:
I don't have the full context here, but the ordering on the `gkeys` does matter for the overlay schedule. `cardano-ledger-specs` is just using the `Ord` instance on key hashes.

Reply:
I based myself on the following line: https://github.com/input-output-hk/cardano-ledger-specs/pull/1849/files#diff-3e7e2e6fbe599db2044bd7f16e5bf9a9R264
But indeed, good point. @JaredCorduan it would be good if `lookupInOverlaySchedule` took a set of them, instead of a list, that would solve the issue.

Reply:
@mrBliss is it okay if I still lookup elements by index? Such as using `Data.Set.elemAt`? (I would check the size first, though, since `elemAt` is partial.)  the old overlay schedule did indeed depend on this ordering.

Reply:
@mrBliss I am going to merge https://github.com/input-output-hk/cardano-ledger-specs/pull/1849 now, just since it is a bit big and it would be nice to have merged. But I am happy to do a quick follow up PR to make `lookupInOverlaySchedule` take a set of genesis keys once I understand the issue!

Reply:
I've raised https://github.com/input-output-hk/cardano-ledger-specs/issues/1856 for this, but would like to get this PR merged first in order to unblock changes in `cardano-node` for the release.